apps/snitch_core/lib/core/domain/shipment_engine.ex

```diff
@@ -1,66 +1,77 @@
 defmodule Snitch.Domain.ShipmentEngine do
   @moduledoc """
-  Implements functions to handle creation of the optimal shipment.
+  Finds the optimal shipment for a given order.
 
-  > Note 
-    No support for partially fulfilled and backordered packages.
+  The current module returns `shipments` under the following constraints:
+  * Selected `shipments` won't fulfill the same `lineitem`.
+  * Selected `shipments` together would fulfill the entire `order`.
+
+  In case `shipments` don't fulfill the above mentioned constraints 
+  and empty`list` is returned.
+
+  Support for `shipments` partially fulfilling an `order` is yet to be added.
+
+  Constraints to be added in future:
+    * prefer shorter shipping distance.
+    * prefer smaller shipping-cost (to user)
+    * prefer smaller shipping cost (to store owner)
+
+  > Note
+  > - Partial fulfillment refers to a scenario when a lineitem is fulfilled 
+    by two different shipments. 
+  > - The module doesn't take into consideration partial fulfillment and backorders.
   """
 
-  import Aruspex.Problem, only: [new: 0, add_variable: 3, post: 4, post: 3]
+  alias Aruspex.Problem
   alias Snitch.Data.Schema.Order
   alias Aruspex.Strategy.SimulatedAnnealing
 
   @domain [true, false]
 
   @doc """
-  Returns a list of packages fulfilling an order
+  Returns the optimal shipment.
 
   Takes a list of `packages` and an `order` struct.
+  An optimal shipment which fulfills the specified business
+  constraints is returned.
+  At present `shipments` which fulfill the `order` and are mutually
```

Comment:
Confusing wording.
What specified business constraints? There's no way to state these constraints so just state what this does.